### PR TITLE
Unsigned requests

### PR DESCRIPTION
**Pull request by endgame**
Add the ability to make unsigned requests. Tested by performing `s3:GetObject` against a bucket whose policy allowed access from `"Principal": "*"`.

Ping @K0Te , since this may be a better way to handle the `sts:AssumeRoleWithWebIdentity` stuff. It also lets us work towards an `amazonka`/`amazonka-auth` split down the line.

### CODE REVIEW

File: lib/amazonka/src/Amazonka.hs [hunk 1]
```diff
@@ -528,7 +528,7 @@ presign ::
 presign env time expires rq =
   Presign.presignWith
     (appEndo (getDual (Env.envOverride env)))
-    (Env.envAuth env)
+    (fromMaybe (error "Amazonka.presign: Env with no Auth") $ Env.envAuth env)
```
Comment (endgame) [inline lib/amazonka/src/Amazonka.hs:531]: Possibly not the best way to do this? But `Env`s with no credentials seem like they'll be incredibly rare...

Open to suggestions, including returning a `Maybe`, using a type-level tracker of whether we have auth info, or something else entirely.

**Comment by brendanhay** [comment]
Oops 🤦🏽‍♂️

**Comment by brendanhay** [comment]
Kinda annoying it closed the branch when I merged, but understandable I guess - apologies, can you reopen on `main`?

**Comment by endgame** [comment]
Coulda sworn that GH would rebase PRs but anyway it's replicated at #708

**Pull request closed without merging**